### PR TITLE
Reorganize DSSE library

### PR DESCRIPTION
**Pull request by adityasaky**
* Reorganizes source files
* Removes Verify method from EnvelopeSigner -> use EnvelopeVerifier directly -- happy to reverse this if needed
* Renames SignVerifier to SignerVerifier (with an alias for compatibility)

### CODE REVIEW

**Comment by adityasaky** [comment]
@pxp928 @mnm678

File: dsse/envelope.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/*
+Both standard and url encoding are allowed:
+https://github.com/secure-systems-lab/dsse/blob/master/envelope.md
+*/
+func b64Decode(s string) ([]byte, error) {
+	b, err := base64.StdEncoding.DecodeString(s)
+	if err != nil {
+		b, err = base64.URLEncoding.DecodeString(s)
+		if err != nil {
+			return nil, err
```
Comment (pxp928) [inline dsse/envelope.go:48]: nit: more description error via `fmt.Errorf`

File: dsse/sign_test.go [hunk 2]
```diff
@@ -39,13 +39,13 @@ func TestPAE(t *testing.T) {
 	})
 }
 
-type nilsigner int
+type nilsignerverifier int
```
Comment (pxp928) [inline dsse/sign_test.go:42]: nit: camelcase

File: dsse/sign_test.go [hunk 3]
```diff
@@ -59,21 +59,21 @@ func (n nilsigner) Verify(ctx context.Context, data, sig []byte) error {
...
 	return "nil", nil
 }
 
-func (n nilsigner) Public() crypto.PublicKey {
+func (n nilsignerverifier) Public() crypto.PublicKey {
 	return "nil-public"
 }
 
-type nullsigner int
+type nullsignerverifier int
```
Comment (pxp928) [inline dsse/sign_test.go:70]: nit: camelcase

File: dsse/sign_test.go [hunk 4]
```diff
@@ -113,24 +113,24 @@ func (n errsigner) Public() crypto.PublicKey {
 	return "err-public"
 }
 
-type errverifier int
+type errsignerverifier int
```
Comment (pxp928) [inline dsse/sign_test.go:116]: nit: camelcase

File: dsse/sign_test.go [hunk 5]
```diff
@@ -246,14 +246,14 @@ func newEcdsaKey() *ecdsa.PrivateKey {
 	return &private
 }
 
-type EcdsaSigner struct {
+type EcdsaSignerVerifier struct {
```
Comment (pxp928) [inline dsse/sign_test.go:249]: nit: camelcase, also should not be exported.

File: dsse/signerverifier.go [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package dsse
+
+// SignerVerifer provides both the signing and verification interface.
+type SignerVerifier interface {
+	Signer
```
Comment (pxp928) [inline dsse/signerverifier.go:5]: could consolidate the `Signer` and `Verifier` interface here.

**Review by pxp928 — APPROVED** [review]
LGTM, few nits.

File: dsse/signerverifier.go (hunk 6 above)
Comment (adityasaky) [reply dsse/signerverifier.go:5]: Do you mean move Signer and Verifier in here as well and have only the EnvelopeSigner and EnvelopeVerifier implementations in sign.go and verify.go?

Comment (adityasaky) [reply dsse/signerverifier.go:37]: Did that in a separate commit.

**Comment by adityasaky** [comment]
Should we loop someone in on removing `Verify` from `EnvelopeSigner`? https://github.com/secure-systems-lab/go-securesystemslib/pull/42/files#diff-0d0655e2b58474147ccefd8432b7cb0c54ecc818f8a17aeeaf80449302cc9d17L94

**Review by pxp928 — APPROVED** [review]
LGTM

**Comment by adityasaky** [comment]
I'm merging this for now so we can test this and #43 with in-toto-golang via the commits. We can cut a release later and decide if we want to add the Verify interface back.

**Pull request merged**